### PR TITLE
user messages for rules 1011, 1012, 2001, 2002.

### PR DESCRIPTION
**Pull request by harleenkohli**

### CODE REVIEW

File: src/Sarif.Multitool/Rules/RuleResources.resx [hunk 1]
```diff
@@ -140,10 +140,12 @@ URIs must conform to [RFC 3986](https://tools.ietf.org/html/rfc3986). In additio
...
   </data>
   <data name="SARIF2001_TerminateMessagesWithPeriod_Warning_Default_Text" xml:space="preserve">
-    <value>{0}: In rule '{1}', the message with id '{2}' does not end in a period. Write rule messages as complete sentences.</value>
+    <value>{0}: In rule '{1}', the message with id '{2}' does not end in a period. Express plain text rule messages as complete sentences. This guidance does not apply to Markdown messages, which might include formatting that makes the punctuation unnecessary.</value>
   </data>
   <data name="SARIF2001_TerminateMessagesWithPeriod_FullDescription_Text" xml:space="preserve">
-    <value>Write in complete sentences and end each sentence with a period. This guidance does not apply to Markdown messages, which might include formatting that makes the punctuation unnecessary.</value>
+    <value>The properties of a result's 'message' property must be consistent with the properties of the rule that the result refers to.
```
Comment (lgolding) [inline src/Sarif.Multitool/Rules/RuleResources.resx:146]: Wrong message.

Comment (harleenkohli) [reply src/Sarif.Multitool/Rules/RuleResources.resx:146]: thanks! yep, my copy pasta! fixing!


---
In reply to: [449246936](https://github.com/microsoft/sarif-sdk/pull/1960#discussion_r449246936) [](ancestors = 449246936)

File: src/Sarif.Multitool/Rules/RuleResources.resx [hunk 2]
```diff
@@ -308,13 +312,15 @@ Many tool use similar names for 'uriBaseId' symbols. We suggest 'REPOROOT' for t
...
   </data>
   <data name="SARIF1012_MessageArgumentsMustBeConsistentWithRule_FullDescription_Text" xml:space="preserve">
-    <value>Placeholder</value>
+    <value>The properties of a result's 'message' property must be consistent with the properties of the rule that the result refers to.
+
+When a result's 'message' object uses the 'id' and 'arguments' properties (which, by the way, is recommended: see SARIF), it must ensure that the rule actually defines a message string with that id, and that 'arguments' array has enough elements to provide values for every replacement sequence in the message specified by 'id'. For example, if the highest numbered replacement sequence in the specified message string is '{3}', then the 'arguments' array must contain at least 4 elements.</value>
```
Comment (lgolding) [inline src/Sarif.Multitool/Rules/RuleResources.resx:323]: >SARIF [](start = 114, length = 5)

The link to the section is missing. Was it not in the document?

Comment (harleenkohli) [reply src/Sarif.Multitool/Rules/RuleResources.resx:323]: yes, this is what i see in md, let me cross check if i have latest version


---
In reply to: [449255220](https://github.com/microsoft/sarif-sdk/pull/1960#discussion_r449255220) [](ancestors = 449255220)

Comment (lgolding) [reply src/Sarif.Multitool/Rules/RuleResources.resx:323]: Yes, my mistake, stand by for fix.

---
In reply to: [449255884](https://github.com/microsoft/sarif-sdk/pull/1960#discussion_r449255884) [](ancestors = 449255884,449255220)

**Pull request merged**